crates/rome_playground/src/App.tsx

```diff
@@ -113,8 +124,8 @@ function App() {
             <div className="w-1/2 p-5">
               <CodeEditor
                 value={code}
-                language="js"
-                placeholder="Enter JS here"
+                language={language}
```

Comment:
It's better to compute information that can be derived from other values than storing it in another state and trying to keep it up to date
```js
function getLanguage(isTypeScript, isJsx): "js" | "jsx" | "typescript" {
    if (isTypeScript) {
      return "typescript";
    } else if (isJsx) {
      return "jsx";
    } else {
      return "js"
    }
  }, [isJsx, isTypeScript]);
```

```suggestion
                language={getLanguage(isTypeScript, isJsx)}
```

Reply:
This is micro optimization :)

Also, with your suggestion the function `getLanguage` gets called even when `isTypeScript` or `isJsx` don't change, at every re-render.

Reply:
Yes, but computing this is cheap. Also, having state out of sync is a common source of problems and it's a common recommendation to recompute it rather than storing it (and if it's expensive, use `useMemo`).

Reply:
>  Also, having state out of sync

I really don't understand what this

Reply:
In case someone forgets to call `setState` after changing some language props. I know, you use `useEffect` but that means that the `language` variable is "stable` until the effect completed. Also be aware, that the hook only runs after render. So react first renders with the default "language" and only then runs the hook.